server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeRequest.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.measure.ws;
+
+import java.util.List;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
+class ComponentTreeRequest {
+
+  private String baseComponentId;
+  private String baseComponentKey;
+  private String component;
+  private String branch;
+  private String strategy;
+  private List<String> qualifiers;
+  private List<String> additionalFields;
+  private String query;
+  private List<String> sort;
+  private Boolean asc;
+  private String metricSort;
+  private Integer metricPeriodSort;
+  private String metricSortFilter;
+  private List<String> metricKeys;
+  private Integer page;
+  private Integer pageSize;
+  private String developerId;
+  private String developerKey;
+
+  /**
+   * @deprecated since 6.6, please use {@link #getComponent()} instead
+   */
+  @Deprecated
+  @CheckForNull
+  public String getBaseComponentId() {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)